src/metabase/db.clj

```diff
@@ -276,30 +277,24 @@
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
 (defn connection-pool
-  "Create a C3P0 connection pool for the given database SPEC."
+  "Create a HikariCP connection pool for the given database SPEC."
   [{:keys [subprotocol subname classname minimum-pool-size idle-connection-test-period excess-timeout]
     :or   {minimum-pool-size           3
            idle-connection-test-period 0
            excess-timeout              (* 30 60)}
     :as   spec}]
-  {:datasource (doto (ComboPooledDataSource.)
-                 (.setDriverClass                  classname)
-                 (.setJdbcUrl                      (str "jdbc:" subprotocol ":" subname))
-                 (.setMaxIdleTimeExcessConnections excess-timeout)
-                 (.setMaxIdleTime                  (* 3 60 60))
-                 (.setInitialPoolSize              3)
-                 (.setMinPoolSize                  minimum-pool-size)
-                 (.setMaxPoolSize                  15)
-                 (.setIdleConnectionTestPeriod     idle-connection-test-period)
-                 (.setTestConnectionOnCheckin      false)
-                 (.setTestConnectionOnCheckout     false)
-                 (.setPreferredTestQuery           nil)
-                 (.setProperties                   (u/prog1 (Properties.)
-                                                     (doseq [[k v] (dissoc spec :classname :subprotocol :subname
-                                                                                :naming :delimiters :alias-delimiter
-                                                                                :excess-timeout :minimum-pool-size
-                                                                                :idle-connection-test-period)]
-                                                       (.setProperty <> (name k) (str v))))))})
+  {:datasource (HikariDataSource.
+                (doto (HikariConfig.)
+                  (.setDriverClassName              classname)
+                  (.setJdbcUrl                      (str "jdbc:" subprotocol ":" subname))
+                  (.setMaxLifetime                  (* 3 60 60))
```

Comment:
I don't think we want this. From reading the HikariCP docs, this is how long a connection lives, which is a combination of time it was in use and idle time. I think that means that every 10.8 seconds (or maybe a bit longer if it's in use for longer than 10.8 seconds) it will discard the connection and make a new one, causing a lot of connection churn. The default of 30 minutes seems like a good place to start, so the line can just be removed.

Reply:
oh you got me there. I had in mind to change it and just forgot about it. Was too excited that it worked 😄